package.json

```diff
@@ -2,61 +2,75 @@
   "name": "bind-event-listener",
   "version": "3.0.0",
   "private": false,
-  "description": "Making binding and unbinding DOM events easier",
+  "description": "Safe DOM Event binding and unbinding",
   "author": "Alex Reardon <alexreardon@gmail.com>",
   "license": "MIT",
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/alexreardon/bind-event-listener.git"
-  },
   "keywords": [
+    "event listener",
+    "event handler",
+    "dom events",
     "dom",
     "ui",
-    "event listener",
-    "event handler"
+    "bind"
   ],
-  "main": "dist/index.js",
-  "types": "dist/index.d.ts",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/alexreardon/bind-event-listener.git"
+  },
   "sideEffects": false,
+  "type": "module",
+  "exports": {
+    "import": "./dist/esm/index.js",
+    "require": "./dist/cjs/index.cjs",
+    "types": "./dist/esm/index.d.ts"
```

Comment:
there are two problems here:
1. conditions are `exports` are matched from the top to the bottom. It doesn't make sense to have `types` last here. This condition should usually be first. TS will still try to match `import` and `require` conditions when appropriate and it might never reach `types` in a manifest like this (it would reach it here as a sibling `.d.ts` file of your `import` condition but that's coincidental)
2. TS will still think that this package can't be required. If you are shipping `.cjs` file here then you should also consider shipping `.d.cts` file and create conditions structure like `{ import: { types, default }, require: { types, default } }`

Reply:
Awesome thanks. Do you know how I can get tsc to generate .d.cts?

Reply:
You can't 😂 Or rather, it generates its outputs based on your inputs. Your input is a `.ts` file (not an `.cts` file) so it won't guess that you want some extra `.d.cts` file here. The thing to understand here is that TS won't really understand that you are trying to ship a dual package from a single set of sources so it won't help you anyhow to generate multiple outputs based on a single input.

You can either `cp && mv` your `.d.ts` or use some tool that handles this for you (tsup or tshy)

Reply:
I think i've done it! I am now using `tsup`. It looks like the declaration files for esm and tsc are the same (which is what I was getting caught on before - i thought the cjs ones needed to be formatted differently). 

Changes:

- now using `tsup` to build `dist` (and declaration files)
- "exports" now pointing to the correct types for given bundles (might not be strictly needed, but seems helpful)
- added `@arethetypeswrong/cli` to validate that the package is setup correctly

Reply:
Keen to know what you think @Andarist

Reply:
[bind-event-listener-3.0.0.tgz](https://github.com/alexreardon/bind-event-listener/files/15449912/bind-event-listener-3.0.0.tgz)

Attached the `npm pack` for reference

Reply:
[bind-event-listener-3.0.0.tgz](https://github.com/alexreardon/bind-event-listener/files/15450665/bind-event-listener-3.0.0.tgz)

Now with tsup workaround: https://github.com/egoist/tsup/issues/953#issuecomment-2132596525